### ISSUE TITLE
4.9.79 kernel crash on GCP

### ISSUE THREAD

**Issue by rn** [linuxkit/linuxkit#2903]
In the kernel update in https://github.com/linuxkit/linuxkit/pull/2900 the GCP test consistently fails with a 4.9.79 kernel crash. This issue is to track progress/debug info etc.
```
[    1.591435] divide error: 0000 [#1] SMP
[    1.591961] Modules linked in:
[    1.592546] CPU: 0 PID: 1 Comm: swapper/0 Not tainted 4.9.79-linuxkit #1
[    1.593461] Hardware name: Google Google Compute Engine/Google Compute Engine, BIOS Google 01/01/2011
[    1.595135] task: ffffa01b2b63c040 task.stack: ffffb24f40010000
[    1.596683] RIP: 0010:[<ffffffffb7d88ce3>]  [<ffffffffb7d88ce3>] init_intel_microcode+0x49/0x5a
[    1.598018] RSP: 0000:ffffb24f40013e48  EFLAGS: 00010206
[    1.599016] RAX: 0000000001400000 RBX: 00000000ffffffea RCX: 0000000000000000
[    1.600093] RDX: 0000000000000000 RSI: ffffa01b2fffa9a8 RDI: ffffffffb7d8888b
[    1.601222] RBP: 00000000ffffffff R08: ffffa01b2fffa9a1 R09: 000000000000005f
[    1.602330] R10: 0000000000000067 R11: ffffa01b22164177 R12: 0000000000000000
[    1.603291] R13: ffffffffb7d787b6 R14: 0000000000000000 R15: 0000000000000000
[    1.604406] FS:  0000000000000000(0000) GS:ffffa01b2fc00000(0000) knlGS:0000000000000000
[    1.606154] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[    1.607493] CR2: 0000000000000000 CR3: 000000002ac0a000 CR4: 0000000000040630
[    1.609310] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[    1.611122] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[    1.613008] Stack:
[    1.613445]  ffffffffb7d888c1 0000000000000000 ffffffffb7d1ba40 0000000000000000
[    1.615423]  0000000000000000 ffffffffb7d787b6 0000000000000000 0000000000000000
[    1.617049]  ffffffffb74fed89 00000000fffffff4 00000000ffffffff 4eae7bc133547c47
[    1.617904] Call Trace:
[    1.618198]  [<ffffffffb7d888c1>] ? microcode_init+0x36/0x1de
[    1.619264]  [<ffffffffb7d787b6>] ? set_debug_rodata+0xc/0xc
[    1.620143]  [<ffffffffb74fed89>] ? driver_register+0xaf/0xb4
[    1.621841]  [<ffffffffb7d8888b>] ? save_microcode_in_initrd+0x3c/0x3c
[    1.623424]  [<ffffffffb70021b7>] ? do_one_initcall+0x98/0x130
[    1.624659]  [<ffffffffb7d787b6>] ? set_debug_rodata+0xc/0xc
[    1.626303]  [<ffffffffb7d79091>] ? kernel_init_freeable+0x166/0x1e7
[    1.627674]  [<ffffffffb77ce3d7>] ? rest_init+0x6e/0x6e
[    1.628942]  [<ffffffffb77ce3e1>] ? kernel_init+0xa/0xe6
[    1.630243]  [<ffffffffb77d8587>] ? ret_from_fork+0x57/0x70
[    1.631592] Code: 16 0f b6 35 a0 ac fb ff 48 c7 c7 c4 ea 99 b7 e8 82 3e 41 ff 31 c0 c3 8b 05 3b ad fb ff 0f b7 0d 54 ad fb ff 31 d2 c1 e0 0a 48 98 <48> f7 f1 89 05 f4 6a 14 00 48 c7 c0 00 d8 c2 b7 c3 4c 8d 54 24 
[    1.638875] RIP  [<ffffffffb7d88ce3>] init_intel_microcode+0x49/0x5a
[    1.640382]  RSP <ffffb24f40013e48>
[    1.641320] ---[ end trace e88ef332e19594b9 ]---
[    1.642418] Kernel panic - not syncing: Fatal exception
[    1.644581] Kernel Offset: 0x36000000 from 0xffffffff81000000 (relocation range: 0xffffffff80000000-0xffffffffbfffffff)
[    1.647312] ---[ end Kernel panic - not syncing: Fatal exception
```

And here the full boot log:
```
[    0.000000] Linux version 4.9.79-linuxkit (root@97758fa53208) (gcc version 6.4.0 (Alpine 6.4.0) ) #1 SMP Thu Feb 1 14:59:17 UTC 2018
[    0.000000] Command line: BOOT_IMAGE=/kernel console=ttyS0 initrd=/initrd.img
[    0.000000] x86/fpu: Supporting XSAVE feature 0x001: 'x87 floating point registers'
[    0.000000] x86/fpu: Supporting XSAVE feature 0x002: 'SSE registers'
[    0.000000] x86/fpu: Supporting XSAVE feature 0x004: 'AVX registers'
[    0.000000] x86/fpu: xstate_offset[2]:  576, xstate_sizes[2]:  256
[    0.000000] x86/fpu: Enabled xstate features 0x7, context size is 832 bytes, using 'standard' format.
[    0.000000] x86/fpu: Using 'eager' FPU context switches.
[    0.000000] e820: BIOS-provided physical RAM map:
[    0.000000] BIOS-e820: [mem 0x0000000000000000-0x000000000009fbff] usable
[    0.000000] BIOS-e820: [mem 0x000000000009fc00-0x000000000009ffff] reserved
[    0.000000] BIOS-e820: [mem 0x00000000000f0000-0x00000000000fffff] reserved
[    0.000000] BIOS-e820: [mem 0x0000000000100000-0x00000000bfff2fff] usable
[    0.000000] BIOS-e820: [mem 0x00000000bfff3000-0x00000000bfffffff] reserved
[    0.000000] BIOS-e820: [mem 0x00000000fffbc000-0x00000000ffffffff] reserved
[    0.000000] BIOS-e820: [mem 0x0000000100000000-0x000000012fffffff] usable
[    0.000000] NX (Execute Disable) protection: active
[    0.000000] SMBIOS 2.4 present.
[    0.000000] Hypervisor detected: KVM
[    0.000000] e820: last_pfn = 0x130000 max_arch_pfn = 0x400000000
[    0.000000] x86/PAT: Configuration [0-7]: WB  WC  UC- UC  WB  WC  UC- WT  
[    0.000000] e820: last_pfn = 0xbfff3 max_arch_pfn = 0x400000000
[    0.000000] found SMP MP-table at [mem 0x000f28d0-0x000f28df] mapped at [ffffa01a000f28d0]
[    0.000000] Using GB pages for direct mapping
[    0.000000] RAMDISK: [mem 0x7dc27000-0x7fffffff]
[    0.000000] ACPI: Early table checksum verification disabled
[    0.000000] ACPI: RSDP 0x00000000000F2890 000014 (v00 Google)
[    0.000000] ACPI: RSDT 0x00000000BFFF3430 000038 (v01 Google GOOGRSDT 00000001 GOOG 00000001)
[    0.000000] ACPI: FACP 0x00000000BFFFCF60 0000F4 (v02 Google GOOGFACP 00000001 GOOG 00000001)
[    0.000000] ACPI: DSDT 0x00000000BFFF3470 0017B2 (v01 Google GOOGDSDT 00000001 GOOG 00000001)
[    0.000000] ACPI: FACS 0x00000000BFFFCF00 000040
[    0.000000] ACPI: FACS 0x00000000BFFFCF00 000040
[    0.000000] ACPI: SSDT 0x00000000BFFF65F0 00690D (v01 Google GOOGSSDT 00000001 GOOG 00000001)
[    0.000000] ACPI: APIC 0x00000000BFFF5D10 00006E (v01 Google GOOGAPIC 00000001 GOOG 00000001)
[    0.000000] ACPI: WAET 0x00000000BFFF5CE0 000028 (v01 Google GOOGWAET 00000001 GOOG 00000001)
[    0.000000] ACPI: SRAT 0x00000000BFFF4C30 0000B8 (v01 Google GOOGSRAT 00000001 GOOG 00000001)
[    0.000000] kvm-clock: Using msrs 4b564d01 and 4b564d00
[    0.000000] kvm-clock: cpu 0, msr 1:2fffc001, primary cpu clock
[    0.000000] kvm-clock: using sched offset of 1324395370 cycles
[    0.000000] clocksource: kvm-clock: mask: 0xffffffffffffffff max_cycles: 0x1cd42e4dffb, max_idle_ns: 881590591483 ns
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x0000000000001000-0x0000000000ffffff]
[    0.000000]   DMA32    [mem 0x0000000001000000-0x00000000ffffffff]
[    0.000000]   Normal   [mem 0x0000000100000000-0x000000012fffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000001000-0x000000000009efff]
[    0.000000]   node   0: [mem 0x0000000000100000-0x00000000bfff2fff]
[    0.000000]   node   0: [mem 0x0000000100000000-0x000000012fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000001000-0x000000012fffffff]
[    0.000000] ACPI: PM-Timer IO Port: 0xb008
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0xff] dfl dfl lint[0x1])
[    0.000000] IOAPIC[0]: apic_id 0, version 17, address 0xfec00000, GSI 0-23
[    0.000000] ACPI: INT_SRC_OVR (bus 0 bus_irq 5 global_irq 5 high level)
[    0.000000] ACPI: INT_SRC_OVR (bus 0 bus_irq 9 global_irq 9 high level)
[    0.000000] ACPI: INT_SRC_OVR (bus 0 bus_irq 10 global_irq 10 high level)
[    0.000000] ACPI: INT_SRC_OVR (bus 0 bus_irq 11 global_irq 11 high level)
[    0.000000] Using ACPI (MADT) for SMP configuration information
[    0.000000] smpboot: Allowing 1 CPUs, 0 hotplug CPUs
[    0.000000] e820: [mem 0xc0000000-0xfffbbfff] available for PCI devices
[    0.000000] Booting paravirtualized kernel on KVM
[    0.000000] clocksource: refined-jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.000000] setup_percpu: NR_CPUS:128 nr_cpumask_bits:128 nr_cpu_ids:1 nr_node_ids:1
[    0.000000] percpu: Embedded 36 pages/cpu @ffffa01b2fc00000 s106904 r8192 d32360 u2097152
[    0.000000] Built 1 zonelists in Zone order, mobility grouping on.  Total pages: 967548
[    0.000000] Kernel command line: BOOT_IMAGE=/kernel console=ttyS0 initrd=/initrd.img
[    0.000000] PID hash table entries: 4096 (order: 3, 32768 bytes)
[    0.000000] Dentry cache hash table entries: 524288 (order: 10, 4194304 bytes)
[    0.000000] Inode-cache hash table entries: 262144 (order: 9, 2097152 bytes)
[    0.000000] Memory: 3745452K/3931716K available (8052K kernel code, 1391K rwdata, 2700K rodata, 1400K init, 564K bss, 186264K reserved, 0K cma-reserved)
[    0.000000] Kernel/User page tables isolation: enabled
[    0.000000] Hierarchical RCU implementation.
[    0.000000] 	Build-time adjustment of leaf fanout to 64.
[    0.000000] 	RCU restricting CPUs from NR_CPUS=128 to nr_cpu_ids=1.
[    0.000000] RCU: Adjusting geometry for rcu_fanout_leaf=64, nr_cpu_ids=1
[    0.000000] NR_IRQS:8448 nr_irqs:256 16
[    0.000000] Console: colour *CGA 80x25
[    0.000000] console [ttyS0] enabled
[    0.000000] tsc: Detected 2600.000 MHz processor
[    0.197083] Calibrating delay loop (skipped) preset value.. 5200.00 BogoMIPS (lpj=26000000)
[    0.198302] pid_max: default: 32768 minimum: 301
[    0.198964] ACPI: Core revision 20160831
[    0.206557] ACPI: 2 ACPI AML tables successfully acquired and loaded
[    0.207643] Security Framework initialized
[    0.208256] Yama: becoming mindful.
[    0.208763] Mount-cache hash table entries: 8192 (order: 4, 65536 bytes)
[    0.209727] Mountpoint-cache hash table entries: 8192 (order: 4, 65536 bytes)
[    0.211643] CPU: Physical Processor ID: 0
[    0.212370] Last level iTLB entries: 4KB 512, 2MB 8, 4MB 8
[    0.213255] Last level dTLB entries: 4KB 512, 2MB 32, 4MB 32, 1GB 0
[    0.214607] Spectre V2 mitigation: Vulnerable: Minimal generic ASM retpoline
[    0.225396] Freeing SMP alternatives memory: 20K
[    0.227977] ftrace: allocating 35563 entries in 139 pages
[    0.273941] smpboot: x86_max_cores == zero !?!?[    0.274645] smpboot: Max logical packages: 1
[    0.275741] x2apic enabled
[    0.277119] Switched APIC routing to physical x2apic.
[    0.280811] ..TIMER: vector=0x30 apic1=0 pin1=0 apic2=-1 pin2=-1
[    0.396430] smpboot: CPU0: Intel(R) Xeon(R) CPU @ 2.60GHz (family: 0x6, model: 0x2d, stepping: 0x7)
[    0.398078] Performance Events: unsupported p6 CPU model 45 no PMU driver, software events only.
[    0.399627] x86: Booted up 1 node, 1 CPUs
[    0.400453] smpboot: Total of 1 processors activated (5200.00 BogoMIPS)
[    0.402557] devtmpfs: initialized
[    0.403123] x86/mm: Memory block size: 128MB
[    0.407298] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.408821] futex hash table entries: 256 (order: 2, 16384 bytes)
[    0.409996] NET: Registered protocol family 16
[    0.410910] cpuidle: using governor ladder
[    0.411510] cpuidle: using governor menu
[    0.412092] ACPI: bus type PCI registered
[    0.412772] PCI: Using configuration type 1 for base access
[    0.415089] HugeTLB registered 1 GB page size, pre-allocated 0 pages
[    0.416043] HugeTLB registered 2 MB page size, pre-allocated 0 pages
[    0.418196] ACPI: Added _OSI(Module Device)
[    0.419846] ACPI: Added _OSI(Processor Device)
[    0.420699] ACPI: Added _OSI(3.0 _SCP Extensions)
[    0.421340] ACPI: Added _OSI(Processor Aggregator Device)
[    0.422527] ACPI: Executed 2 blocks of module-level executable AML code
[    0.446208] ACPI: Interpreter enabled
[    0.447118] ACPI: (supports S0 S5)
[    0.447655] ACPI: Using IOAPIC for interrupt routing
[    0.448384] PCI: Using host bridge windows from ACPI; if necessary, use "pci=nocrs" and report a bug
[    0.476534] ACPI: PCI Root Bridge [PCI0] (domain 0000 [bus 00-ff])
[    0.478424] acpi PNP0A03:00: _OSC: OS supports [ASPM ClockPM Segments MSI]
[    0.480602] acpi PNP0A03:00: _OSC failed (AE_NOT_FOUND); disabling ASPM
[    0.481855] acpi PNP0A03:00: fail to add MMCONFIG information, can't access extended PCI configuration space under this bridge.
[    0.484134] PCI host bridge to bus 0000:00
[    0.485276] pci_bus 0000:00: root bus resource [io  0x0000-0x0cf7 window]
[    0.487375] pci_bus 0000:00: root bus resource [io  0x0d00-0xffff window]
[    0.488498] pci_bus 0000:00: root bus resource [mem 0x000a0000-0x000bffff window]
[    0.490028] pci_bus 0000:00: root bus resource [mem 0xc0000000-0xfebfffff window]
[    0.491215] pci_bus 0000:00: root bus resource [bus 00-ff]
[    0.539918] pci 0000:00:01.3: quirk: [io  0xb000-0xb03f] claimed by PIIX4 ACPI
[    0.586278] ACPI: PCI Interrupt Link [LNKA] (IRQs 5 *10 11)
[    0.588430] ACPI: PCI Interrupt Link [LNKB] (IRQs 5 *10 11)
[    0.591277] ACPI: PCI Interrupt Link [LNKC] (IRQs 5 10 *11)
[    0.593366] ACPI: PCI Interrupt Link [LNKD] (IRQs 5 10 *11)
[    0.595467] ACPI: PCI Interrupt Link [LNKS] (IRQs *9)
[    0.612646] ACPI: Enabled 16 GPEs in block 00 to 0F
[    0.613591] SCSI subsystem initialized
[    0.614174] ACPI: bus type USB registered
[    0.614961] usbcore: registered new interface driver usbfs
[    0.615823] usbcore: registered new interface driver hub
[    0.616648] usbcore: registered new device driver usb
[    0.617419] pps_core: LinuxPPS API ver. 1 registered
[    0.618556] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.619899] PTP clock support registered
[    0.620537] PCI: Using ACPI for IRQ routing
[    0.621378] NetLabel: Initializing
[    0.621981] NetLabel:  domain hash size = 128
[    0.622679] NetLabel:  protocols = UNLABELED CIPSOv4
[    0.623774] NetLabel:  unlabeled traffic allowed by default
[    0.626758] wmi: Mapper loaded
[    0.627884] clocksource: Switched to clocksource kvm-clock
[    0.641807] VFS: Disk quotas dquot_6.6.0
[    0.642572] VFS: Dquot-cache hash table entries: 512 (order 0, 4096 bytes)
[    0.644560] FS-Cache: Loaded
[    0.645063] CacheFiles: Loaded
[    0.645620] pnp: PnP ACPI init
[    0.646668] pnp: PnP ACPI: found 7 devices
[    0.654005] clocksource: acpi_pm: mask: 0xffffff max_cycles: 0xffffff, max_idle_ns: 2085701024 ns
[    0.655548] NET: Registered protocol family 2
[    0.656314] TCP established hash table entries: 32768 (order: 6, 262144 bytes)
[    0.657562] TCP bind hash table entries: 32768 (order: 7, 524288 bytes)
[    0.659177] TCP: Hash tables configured (established 32768 bind 32768)
[    0.660232] UDP hash table entries: 2048 (order: 4, 65536 bytes)
[    0.661171] UDP-Lite hash table entries: 2048 (order: 4, 65536 bytes)
[    0.662119] NET: Registered protocol family 1
[    0.662737] pci 0000:00:00.0: Limiting direct PCI/PCI transfers
[    0.664269] Unpacking initramfs...
[    1.286654] Freeing initrd memory: 36708K
[    1.287647] PCI-DMA: Using software bounce buffering for IO (SWIOTLB)
[    1.288679] software IO TLB [mem 0xbbff3000-0xbfff3000] (64MB) mapped at [ffffa01abbff3000-ffffa01abfff2fff]
[    1.290288] RAPL PMU: API unit is 2^-32 Joules, 3 fixed counters, 10737418240 ms ovfl timer
[    1.291508] RAPL PMU: hw unit of domain pp0-core 2^-0 Joules
[    1.292513] RAPL PMU: hw unit of domain package 2^-0 Joules
[    1.293327] RAPL PMU: hw unit of domain dram 2^-0 Joules
[    1.295335] AVX2 or AES-NI instructions are not detected.
[    1.296180] AVX2 instructions are not detected.
[    1.297228] audit: initializing netlink subsys (disabled)
[    1.298254] audit: type=2000 audit(1517677522.724:1): initialized
[    1.299660] workingset: timestamp_bits=46 max_order=20 bucket_order=0
[    1.301280] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    1.302912] FS-Cache: Netfs 'cifs' registered for caching
[    1.303701] fuse init (API version 7.26)
[    1.304397] SGI XFS with ACLs, security attributes, no debug enabled
[    1.305666] 9p: Installing v9fs 9p2000 file system support
[    1.306500] FS-Cache: Netfs '9p' registered for caching
[    1.309041] NET: Registered protocol family 38
[    1.309793] Key type asymmetric registered
[    1.310355] Asymmetric key parser 'x509' registered
[    1.311294] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 251)
[    1.312428] io scheduler noop registered
[    1.313075] io scheduler deadline registered (default)
[    1.314001] io scheduler cfq registered
[    1.314669] pci_hotplug: PCI Hot Plug PCI Core version: 0.5
[    1.315539] pciehp: PCI Express Hot Plug Controller Driver version: 0.4
[    1.316682] shpchp: Standard Hot Plug PCI Controller Driver version: 0.4
[    1.317735] hv_vmbus: registering driver hyperv_fb
[    1.318685] input: Power Button as /devices/LNXSYSTM:00/LNXPWRBN:00/input/input0
[    1.320066] ACPI: Power Button [PWRF]
[    1.321162] input: Sleep Button as /devices/LNXSYSTM:00/LNXSLPBN:00/input/input1
[    1.322355] ACPI: Sleep Button [SLPF]
[    1.323197] GHES: HEST is not enabled!
[    1.325233] ACPI: PCI Interrupt Link [LNKC] enabled at IRQ 11
[    1.326269] virtio-pci 0000:00:03.0: virtio_pci: leaving for legacy driver
[    1.331674] ACPI: PCI Interrupt Link [LNKD] enabled at IRQ 10
[    1.332562] virtio-pci 0000:00:04.0: virtio_pci: leaving for legacy driver
[    1.337195] xenfs: not registering filesystem on non-xen platform
[    1.338508] Serial: 8250/16550 driver, 4 ports, IRQ sharing disabled
[    1.361327] 00:03: ttyS0 at I/O 0x3f8 (irq = 4, base_baud = 115200) is a 16550A
[    1.385135] 00:04: ttyS1 at I/O 0x2f8 (irq = 3, base_baud = 115200) is a 16550A
[    1.408315] 00:05: ttyS2 at I/O 0x3e8 (irq = 6, base_baud = 115200) is a 16550A
[    1.430657] 00:06: ttyS3 at I/O 0x2e8 (irq = 7, base_baud = 115200) is a 16550A
[    1.432226] Non-volatile memory driver v1.3
[    1.433518] Hangcheck: starting hangcheck timer 0.9.1 (tick is 180 seconds, margin is 60 seconds).
[    1.436182] loop: module loaded
[    1.438234] nbd: registered device at major 43
[    1.450496] scsi host0: Virtio SCSI HBA
[    1.452849] scsi 0:0:1:0: Direct-Access     Google   PersistentDisk   1    PQ: 0 ANSI: 6
[    1.472831] random: fast init done
[    1.474974] VMware PVSCSI driver - version 1.0.7.0-k
[    1.475965] hv_vmbus: registering driver hv_storvsc
[    1.477214] sd 0:0:1:0: [sda] 2097152 512-byte logical blocks: (1.07 GB/1.00 GiB)
[    1.478592] sd 0:0:1:0: [sda] 4096-byte physical blocks
[    1.480952] sd 0:0:1:0: Attached scsi generic sg0 type 0
[    1.481821] sd 0:0:1:0: [sda] Write Protect is off
[    1.483971] tun: Universal TUN/TAP device driver, 1.6
[    1.484940] tun: (C) 1999-2004 Max Krasnyansky <maxk@qualcomm.com>
[    1.487250] sd 0:0:1:0: [sda] Write cache: enabled, read cache: enabled, doesn't support DPO or FUA
[    1.526457] VMware vmxnet3 virtual NIC driver - version 1.4.a.0-k-NAPI
[    1.527811] hv_vmbus: registering driver hv_netvsc
[    1.528588] Fusion MPT base driver 3.04.20
[    1.529401] Copyright (c) 1999-2008 LSI Corporation
[    1.530168] Fusion MPT SPI Host driver 3.04.20
[    1.530967] i8042: PNP: PS/2 Controller [PNP0303:KBD,PNP0f13:MOU] at 0x60,0x64 irq 1,12
[    1.532922] i8042: Warning: Keylock active
[    1.534395]  sda: sda1
[    1.535341] serio: i8042 KBD port at 0x60,0x64 irq 1
[    1.536117] serio: i8042 AUX port at 0x60,0x64 irq 12
[    1.537123] hv_vmbus: registering driver hyperv_keyboard
[    1.537846] mousedev: PS/2 mouse device common for all mice
[    1.538793] sd 0:0:1:0: [sda] Attached SCSI disk
[    1.539568] input: PC Speaker as /devices/platform/pcspkr/input/input3
[    1.541295] rtc_cmos 00:00: RTC can wake from S4
[    1.542680] rtc_cmos 00:00: rtc core: registered rtc_cmos as rtc0
[    1.544143] rtc_cmos 00:00: alarms up to one day, 114 bytes nvram
[    1.545128] i2c /dev entries driver
[    1.545922] device-mapper: ioctl: 4.35.0-ioctl (2016-06-23) initialised: dm-devel@redhat.com
[    1.549697] usbcore: registered new interface driver usbhid
[    1.550413] usbhid: USB HID core driver
[    1.551002] hv_utils: Registering HyperV Utility Driver
[    1.551666] hv_vmbus: registering driver hv_util
[    1.552592] hv_vmbus: registering driver hv_balloon
[    1.553224] oprofile: using timer interrupt.
[    1.554284] GACT probability on
[    1.555069] Mirror/redirect action on
[    1.556080] Simple TC action Loaded
[    1.556481] u32 classifier
[    1.556847]     Performance counters on
[    1.557313]     input device check on
[    1.557856]     Actions configured
[    1.558569] Netfilter messages via NETLINK v0.30.
[    1.559153] nfnl_acct: registering with nfnetlink.
[    1.559770] nf_conntrack version 0.5.0 (16384 buckets, 65536 max)
[    1.560464] ctnetlink v0.93: registering with nfnetlink.
[    1.561472] nf_tables: (c) 2007-2009 Patrick McHardy <kaber@trash.net>
[    1.562968] nf_tables_compat: (c) 2012 Pablo Neira Ayuso <pablo@netfilter.org>
[    1.564580] xt_time: kernel timezone is -0000
[    1.565117] ip_set: protocol 6
[    1.565457] IPVS: Registered protocols (TCP, UDP, SCTP, AH, ESP)
[    1.566630] IPVS: Connection hash table configured (size=4096, memory=64Kbytes)
[    1.567582] IPVS: Creating netns size=2104 id=0
[    1.568684] IPVS: ipvs loaded.
[    1.569437] IPVS: [rr] scheduler registered.
[    1.570109] IPVS: [wrr] scheduler registered.
[    1.570871] IPVS: [lc] scheduler registered.
[    1.571437] IPVS: [wlc] scheduler registered.
[    1.572038] IPVS: [fo] scheduler registered.
[    1.572643] IPVS: [ovf] scheduler registered.
[    1.573157] IPVS: [lblc] scheduler registered.
[    1.573742] IPVS: [lblcr] scheduler registered.
[    1.574242] IPVS: [dh] scheduler registered.
[    1.575127] IPVS: [sh] scheduler registered.
[    1.576112] IPVS: [sed] scheduler registered.
[    1.576837] IPVS: [nq] scheduler registered.
[    1.577639] IPVS: ftp: loaded support on port[0] = 21
[    1.578474] ipip: IPv4 and MPLS over IPv4 tunneling driver
[    1.579606] gre: GRE over IPv4 demultiplexor driver
[    1.580239] ip_tables: (C) 2000-2006 Netfilter Core Team
[    1.581319] ipt_CLUSTERIP: ClusterIP Version 0.8 loaded successfully
[    1.583104] arp_tables: arp_tables: (C) 2002 David S. Miller
[    1.584274] NET: Registered protocol family 10
[    1.585398] ip6_tables: (C) 2000-2006 Netfilter Core Team
[    1.586848] NET: Registered protocol family 17
[    1.587569] Bridge firewalling registered
[    1.588185] Ebtables v2.0 registered
[    1.588909] 8021q: 802.1Q VLAN Support v1.8
[    1.589795] 9pnet: Installing 9P2000 support
[    1.590406] Key type dns_resolver registered
[    1.591435] divide error: 0000 [#1] SMP
[    1.591961] Modules linked in:
[    1.592546] CPU: 0 PID: 1 Comm: swapper/0 Not tainted 4.9.79-linuxkit #1
[    1.593461] Hardware name: Google Google Compute Engine/Google Compute Engine, BIOS Google 01/01/2011
[    1.595135] task: ffffa01b2b63c040 task.stack: ffffb24f40010000
[    1.596683] RIP: 0010:[<ffffffffb7d88ce3>]  [<ffffffffb7d88ce3>] init_intel_microcode+0x49/0x5a
[    1.598018] RSP: 0000:ffffb24f40013e48  EFLAGS: 00010206
[    1.599016] RAX: 0000000001400000 RBX: 00000000ffffffea RCX: 0000000000000000
[    1.600093] RDX: 0000000000000000 RSI: ffffa01b2fffa9a8 RDI: ffffffffb7d8888b
[    1.601222] RBP: 00000000ffffffff R08: ffffa01b2fffa9a1 R09: 000000000000005f
[    1.602330] R10: 0000000000000067 R11: ffffa01b22164177 R12: 0000000000000000
[    1.603291] R13: ffffffffb7d787b6 R14: 0000000000000000 R15: 0000000000000000
[    1.604406] FS:  0000000000000000(0000) GS:ffffa01b2fc00000(0000) knlGS:0000000000000000
[    1.606154] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[    1.607493] CR2: 0000000000000000 CR3: 000000002ac0a000 CR4: 0000000000040630
[    1.609310] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[    1.611122] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[    1.613008] Stack:
[    1.613445]  ffffffffb7d888c1 0000000000000000 ffffffffb7d1ba40 0000000000000000
[    1.615423]  0000000000000000 ffffffffb7d787b6 0000000000000000 0000000000000000
[    1.617049]  ffffffffb74fed89 00000000fffffff4 00000000ffffffff 4eae7bc133547c47
[    1.617904] Call Trace:
[    1.618198]  [<ffffffffb7d888c1>] ? microcode_init+0x36/0x1de
[    1.619264]  [<ffffffffb7d787b6>] ? set_debug_rodata+0xc/0xc
[    1.620143]  [<ffffffffb74fed89>] ? driver_register+0xaf/0xb4
[    1.621841]  [<ffffffffb7d8888b>] ? save_microcode_in_initrd+0x3c/0x3c
[    1.623424]  [<ffffffffb70021b7>] ? do_one_initcall+0x98/0x130
[    1.624659]  [<ffffffffb7d787b6>] ? set_debug_rodata+0xc/0xc
[    1.626303]  [<ffffffffb7d79091>] ? kernel_init_freeable+0x166/0x1e7
[    1.627674]  [<ffffffffb77ce3d7>] ? rest_init+0x6e/0x6e
[    1.628942]  [<ffffffffb77ce3e1>] ? kernel_init+0xa/0xe6
[    1.630243]  [<ffffffffb77d8587>] ? ret_from_fork+0x57/0x70
[    1.631592] Code: 16 0f b6 35 a0 ac fb ff 48 c7 c7 c4 ea 99 b7 e8 82 3e 41 ff 31 c0 c3 8b 05 3b ad fb ff 0f b7 0d 54 ad fb ff 31 d2 c1 e0 0a 48 98 <48> f7 f1 89 05 f4 6a 14 00 48 c7 c0 00 d8 c2 b7 c3 4c 8d 54 24 
[    1.638875] RIP  [<ffffffffb7d88ce3>] init_intel_microcode+0x49/0x5a
[    1.640382]  RSP <ffffb24f40013e48>
[    1.641320] ---[ end trace e88ef332e19594b9 ]---
[    1.642418] Kernel panic - not syncing: Fatal exception
[    1.644581] Kernel Offset: 0x36000000 from 0xffffffff81000000 (relocation range: 0xffffffff80000000-0xffffffffbfffffff)
[    1.647312] ---[ end Kernel panic - not syncing: Fatal exception
```

**Comment by rn** [comment]
The ci boots a `n1-standard-1` machine type in `europe-west1-b` which is a single CPU VM.

From 4.9.78 to 4.9.79 [this patch](https://lkml.org/lkml/2018/1/24/325) was added which adds:
```
do_div(llc_size, c->x86_max_cores)
```
to `init_intel_microcode()` which is called from `microcode_init`.

And indeed:
```
[    0.225396] Freeing SMP alternatives memory: 20K
[    0.227977] ftrace: allocating 35563 entries in 139 pages
[    0.273941] smpboot: x86_max_cores == zero !?!?
[    0.274645] smpboot: Max logical packages: 1
[    0.275741] x2apic enabled
```

If I boot the same kernel on KVM on Linux:
```
[    0.145356] Freeing SMP alternatives memory: 20K
[    0.147227] ftrace: allocating 35761 entries in 140 pages
[    0.190558] smpboot: Max logical packages: 1
[    0.191263] x2apic enabled
```

**Comment by rn** [comment]
I see the same crash for 4.9.79 and 4.4.114 on `g1-small` (also 1 vCPU) but not on `n1-standard-2` (2 vCPUs).

There is no crash with 4.14.16 and 4.15. 

The updates to 4.9.80 and 4.4.115 don't seem to contain a fix either. I'll prepare a patch for stable

**Comment by rn** [comment]
I've just merged a hacky workaround in 

In other news, from Borislav Petkov, one of the kernel maintainers:
```
The microcode loader is not supposed to load in a guest.

Greg, I guess you need something like this:

a15a753539ec ("x86/microcode/AMD: Do not load when running on a hypervisor")

I haven't tried whether it applies cleanly though...
```

I keep this issue open till there is a proper fix upstream and the workaround is removed

**Comment by rn** [comment]
Proper fix should be in the next 4.4.x and 4.9.x update. The patches have been queued up

**Comment by rn** [comment]
4.9.81 has the proper fix (see #2912) but still waiting for the 4.4 backport
